code/ship/shiphit.cpp

```diff
@@ -2386,6 +2386,27 @@ static void ship_do_damage(object *ship_objp, object *other_obj, vec3d *hitpos,
 				damage = Armor_types[shipp->armor_type_idx].GetDamage(damage, damage_type_idx, difficulty_scale_factor, other_obj_is_beam);
 				apply_diff_scale = false;
 			}
+		}
+			// if weapon is vampiric, slap healing on shooter instead of target
+		if (Weapons[other_obj->instance].weapon_info_index >= 0){
+			weapon_info *wip = &Weapon_info[Weapons[other_obj->instance].weapon_info_index];
+			if (wip->wi_flags[Weapon::Info_Flags::Vampiric]){
+				if (other_obj->parent > 0){
+					object* parent = &Objects[other_obj->parent];
+					if (parent->type == OBJ_SHIP) {
+						if (parent->signature == other_obj->parent_sig){
```

Comment:
Same here.  Also you can add `!parent->flags[Object::Object_Flags::Should_be_dead]` to further consolidate things.